src/main.rs

```diff
@@ -58,6 +58,42 @@ use errors::*;
 struct Options<'a> {
     true_color: bool,
     language: Option<&'a str>,
+    interactive_terminal: bool,
+}
+
+enum Shell<'a> {
+    Interactive(Child),
+    NonInteractive(StdoutLock<'a>),
+}
+
+impl<'a> Shell<'a> {
+    fn new(interactive: bool, stdout: &'a Stdout) -> Result<Self> {
```

Comment:
I think I'd prefer if this would be split into two methods (`new_*`) instead of passing a boolean argument. The if-else would have to be at the call site.

What do you think?

Reply:
Sure that makes sense